### PR TITLE
Switch goals to objectives and key results

### LINKED ISSUE #2007: Change goals to objectives and key results
From #1819 #1850. The `Performance Goals and Benchmarks` and section will become objectives and key results. Mockup:

mockup

### This task is done when...
- [x] data model is updated
  - `activity.goals` is currently an array of objects shaped like `{ description: string, objective: string }`
  - `activity.goals` will go away and be replaced with `activity.objectives`, an array shaped like `{ objective: string, keyResults: [ keyResult: string, target: string, baseline: string ]}`
- [x] data is migrated from `activity.goals` to `activity.objectives`
  - `goals[].description` becomes `objectives[].objective`
  - `goals[].objective` becomes `objectives[].keyResults[0].keyResult`
  - `objectives[].keyResults[].target` and `objectives[].keyResults[].target` default to `''`
- [x] frontend is updated to consume the new data shape
- [x] frontend UI is updated

### PR DESCRIPTION
**Pull request by mgwalker**
Switches the activity goal section over to objectives and key results

### This pull request changes...

- migrates existing APDs to the new format (defaults key result baseline and target to `""`)
- updates the backend to put `objectives` into new APDs instead of `goals`
- updates the test seed data, so the tests are good
- updates the UI to display objectives and key results instead of goals
- closes #2007

### This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [x] The change has been documented
  - [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate

### This feature is done when...

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: api/migrations/20200117220657_change-goals-to-okrs.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                  // in the mapped result, goals won't be there. Yay!
+                  ({ goals, ...activity }) => ({
+                    ...activity,
+                    objectives: goals.map(goal => ({
+                      objective: goal.description,
+                      keyResults: [
+                        {
+                          keyResult: goal.objective,
+                          target: '',
+                          baseline: ''
```
Comment (mgwalker) [inline api/migrations/20200117220657_change-goals-to-okrs.js]: To be honest, all this stuff makes me wish Javascript had something more like Python's list comprehensions.

File: api/routes/apds/patch.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ const ajv = new Ajv({
   allErrors: true,
   jsonPointers: true,
   // The validator will remove any fields that aren't in the schema
-  removeAdditional: true
+  removeAdditional: 'all'
```
Comment (mgwalker) [inline api/routes/apds/patch.js]: innocent whistling

**Comment by hbillings** [comment]
@mgwalker I bet @jeromeleecms's bug is related to the one we were discussing the other day, about `Review` components and `autofocus`

**Comment by mgwalker** [comment]
@hbillings Related, yeah. It's because we have an `initialCollapsed` flag, which is based on the assumption that the form/review components will only be loaded once and thereafter just hidden in the DOM. We read that flag on component load and then never again, so everything works if that assumption holds. But that's not how it works now. The modal unloads from the DOM and so every time you open the activity, you will see every newly-added form/review item expanded. It's not unique to the OKRs.

I think we should open a separate issue on that one because it's going to take a little plumbing of every form/review component to fix it up.

**Comment by mgwalker** [comment]
Added #2026

File: web/src/components/FormAndReviewList.test.js [hunk 3]
```diff
@@ -171,6 +172,73 @@ describe('FormAndReviewList component', () => {
...
+            { onClick: btn1, text: 'button 1' },
+            { onClick: btn2, text: 'button 2' }
+          ]}
+          index={17}
+          initialCollapsed={false}
+        />
+      );
+
+      // Find the custom buttons. The "Done" button gets the primary variation,
+      // so exclude that one.
```
Comment (hbillings) [inline web/src/components/FormAndReviewList.test.js]: I am so glad you put this comment here.

File: web/src/containers/activity/ObjectivesAndKeyResults/ObjectiveAndKeyResultForm.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  const changeBaseline = i => ({ target: { value } }) => {
+    setBaseline(activityIndex, index, i, value);
+  };
+
+  return (
+    <Fragment>
+      <TextField
+        autoFocus
+        name="objective"
+        className="data-entry-box"
```
Comment (hbillings) [inline web/src/containers/activity/ObjectivesAndKeyResults/ObjectiveAndKeyResultForm.js]: I really wish I'd come up with a better class name back when.

**Pull request merged**